### ISSUE TITLE
thread_state_container_test fails occasionally

### ISSUE THREAD

**Issue by mbutrovich** [cmu-db/terrier#561]
We're seeing failures rarely on CI for `thread_state_container_test`. I believe @amlatyrngom investigated it briefly and said that it required a large number of local iterations before it manifested as a memory leak.

**Comment by pmenon** [comment]
Do you have a trace? This occasionally happens in the TPL repo and it's a false positive reported in Intel TBB. TBB will statically initialize and allocate data that lives with the process. While technically that memory isn't freed, it is reclaimed by the OS when the process exits. Probably makes sense to add TBB to suppressions.

**Comment by amlatyrngom** [comment]
Error:
```
=================================================================
==14629==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 1560 byte(s) in 3 object(s) allocated from:
    #0 0x7f7445758618 in operator new[](unsigned long) (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xe0618)
    #1 0x7f744545d752  (/usr/lib/x86_64-linux-gnu/libtbb.so.2+0x21752)

SUMMARY: AddressSanitizer: 1560 byte(s) leaked in 3 allocation(s).
Running main() from gmock_main.cc
```

It's looks false positive on the in place `new` at line 52:
```
auto obj = new (s) Object();
```
It can be avoided by using `reinterpret_cast<Object>(s)` instead.

**Comment by apavlo** [comment]
Fixed in #1014
